nav2_core/include/nav2_core/controller.hpp

```diff
@@ -116,6 +116,16 @@ class Controller
     const geometry_msgs::msg::Twist & velocity,
     nav2_core::GoalChecker * goal_checker) = 0;
 
+  /**
+   * @brief Cancel the current control action
+   * @return True if the cancellation was successful. If false is returned, computeVelocityCommands
+   * will be called until cancel returns true.
+   */
+  virtual bool cancel()
```

Comment:
Just make sure to add to the Nav2 tutorial on Controllers in the doc updates

Reply:
https://github.com/ros-planning/navigation.ros.org/pull/535